ArduPlane/mode.cpp

```diff
@@ -125,6 +125,12 @@ bool Mode::enter()
         // but it should be harmless to disable the fence temporarily in these situations as well
         plane.fence.manual_recovery_start();
 #endif
+        //reset mission if in landing sequence, disarmed, not flying, and have changed to a non-autothrottle mode to clear prearm
+        if (plane.mission.get_in_landing_sequence_flag() &&
+            !plane.is_flying() && !plane.arming.is_armed_and_safety_off() &&
+            !plane.throttle_suppressed) {
```

Comment:
why the throttle suppressed check?

Reply:
better to use !plane.control_mode->does_auto_navigation()

Reply:
done